ios-target/ViewModel/Home/TargetFormViewModel.swift

```diff
@@ -77,9 +84,16 @@ class TargetFormViewModel {
                         radius: targetArea,
                         topicId: targetTopic)
     TargetAPI.createTarget(target,
-                           success: { [weak self] target in
+                           success: { [weak self] response in
                             self?.state = .idle
+                            guard let target = response["target"] as? Target else { return }
```

Comment:
you can avoid all this casting by returning a `MatchResponse` on the `createTarget`method